controls/radchartview/populating-with-data/chartdatasource.md

```diff
@@ -64,128 +57,121 @@ We'll create BarSeries with DateTimeCategorical Horizontal Axis, bound via the C
-			''' </summary> '''
-			Public Shared ReadOnly DataProperty As DependencyProperty = DependencyProperty.Register("Data", GetType(ObservableCollection(Of SalesInfo)), GetType(SimpleViewModel), New PropertyMetadata(Nothing))
-	
-			Public Property Data() As RadObservableCollection(Of SalesInfo)
-				Get
-					Return CType(Me.GetValue(DataProperty), RadObservableCollection(Of SalesInfo))
-				End Get
-				Set(ByVal value As RadObservableCollection(Of SalesInfo))
-					Me.SetValue(DataProperty, value)
-				End Set
-			End Property
-	
-			Public Sub New()
-				Dim data = New RadObservableCollection(Of SalesInfo)()
-	
-				Dim startDate As New DateTime(2013, 5, 1)
-	
-				For i As Integer = 0 To 19
-					data.Add(New SalesInfo() With {.Time = startDate.AddDays(i), .Value = i })
-				Next
-	
-				Me.Data = data
-			End Sub
-	 End Class
+	Public Class SimpleViewModel
+	 Inherits DependencyObject
+		   Private r As New Random()
+		''' <summary> '''
+		''' Identifies the <see cref="Data"/> dependency property. '''
+		''' </summary> '''
+		Public Shared ReadOnly DataProperty As DependencyProperty = DependencyProperty.Register("Data", GetType(ObservableCollection(Of SalesInfo)), GetType(SimpleViewModel), New PropertyMetadata(Nothing))
+
+		Public Property Data() As RadObservableCollection(Of SalesInfo)
+			Get
+				Return CType(Me.GetValue(DataProperty), RadObservableCollection(Of SalesInfo))
+			End Get
+			Set(ByVal value As RadObservableCollection(Of SalesInfo))
+				Me.SetValue(DataProperty, value)
+			End Set
+		End Property
+
+		Public Sub New()
+			Dim data = New RadObservableCollection(Of SalesInfo)()
+
+			Dim startDate As New DateTime(2013, 5, 1)
+
+			For i As Integer = 0 To 19
+				data.Add(New SalesInfo() With {.Time = startDate.AddDays(i), .Value = i })
+			Next
+
+			Me.Data = data
+		End Sub
+	End Class
 		
-		Public Class SalesInfo
-	    Dim _time As Date
-	    Dim _value As Int
-	
-	    Public Property [Time]() As DateTime
-	        Get
-	            Return Me._time
-	        End Get
-	        Set(value As DateTime)
-	            Me._time = value
-	        End Set
-	    End Property
-	
-	    Public Property Value() As Int
-	        Get
-	            Return Me._value
-	        End Get
-	        Set(value As Double)
-	            Me._value = value
-	        End Set
-	    End Property
+	Public Class SalesInfo
+		Dim _time As Date
+		Dim _value As Int
+
+		Public Property [Time]() As DateTime
+			Get
+				Return Me._time
+			End Get
+			Set(value As DateTime)
+				Me._time = value
+			End Set
+		End Property
+
+		Public Property Value() As Int
+			Get
+				Return Me._value
+			End Get
+			Set(value As Double)
+				Me._value = value
+			End Set
+		End Property
 	End Class
-	{{endregion}}
-
-
+{{endregion}}
 
 By setting the SamplingUnit to Day and the SamplingUnitInterval to 5 you'll see 4 bars, where each bar repesents a period of 5 days. The result is shown on the image below:

```

Comment:
By setting the SamplingUnit to Day and the SamplingUnitInterval to 5 you'll see 4 bars, where each bar represents a period of 5 days. The result is shown in the image below: